ext/iobuffer/iobuffer.c

```diff
@@ -404,17 +409,22 @@ static VALUE
 IO_Buffer_write_to(VALUE self, VALUE io)
 {
     struct buffer  *buf;
-#if HAVE_RB_IO_T
+#if defined(HAVE_RB_IO_T) || defined(HAVE_RB_IO_DESCRIPTOR)
     rb_io_t        *fptr;
 #else
     OpenFile       *fptr;
 #endif
 
     Data_Get_Struct(self, struct buffer, buf);
-    GetOpenFile(rb_convert_type(io, T_FILE, "IO", "to_io"), fptr);
+    io = rb_convert_type(io, T_FILE, "IO", "to_io");
+    GetOpenFile(io, fptr);
     rb_io_set_nonblock(fptr);
 
+#ifdef HAVE_RB_IO_DESCRIPTOR
+    return INT2NUM(rb_io_descriptor(io));
```

Comment:
This looks wrong to me.

Reply:
Fixed in https://github.com/socketry/cool.io/pull/79/commits/2112022fe0d8f318aeb7412912ecc63a6ba50bbf.